lib/services.js

```diff
@@ -390,8 +391,9 @@ const services = {
         if (params.tagging) {
             multipartObjectMD['x-amz-tagging'] = params.tagging;
         }
-        if (params.retentionInfo) {
-            multipartObjectMD.retentionInfo = params.retentionInfo;
+        if (params.retentionMode && params.retentionDate) {
+            multipartObjectMD.retentionMode = params.retentionMode;
+            multipartObjectMD.retentionDate = params.retentionDate;
```

Comment:
maybe setter could be used here as well; 

```suggestion
            multipartObjectMD. setRetentionMode(retentionMode) = params.retentionMode;
            multipartObjectMD. setRetentionDate(retentionDate) = params.retentionDate;
```

Reply:
Yeah, our use of objectMD as an instance vs as JSON is super inconsistent throughout Cloudserver, so I just followed what was done in the function around it. Definitely something to address though.

Reply:
Oh I see, sure, makes sense. :)